src/test/java/javaslang/collection/AbstractMapTest.java

```diff
@@ -99,6 +99,11 @@
         return IntMap.of(emptyMap());
     }
 
+    @Override
+    protected boolean emptyShouldBeSingleton() {
+        return false;
```

Comment:
Can't we just overwrite the method `emptyShouldBeSingleton` and let `AbstractMapTest` return `true` and `TreeMapTest` return `false`? Then the tests should run, right?